src/AST-Core/RBParser.class.st

```diff
@@ -1234,19 +1232,26 @@ RBParser >> parseVariableNode [
 { #category : #'error handling' }
 RBParser >> parserError: aString [
 	"Let the errorBlock try to recover from the error. Answer a ParseNode (possibly an RBParseErrorNode) or signal there is new source"
-	| errorNode errorMessage errorPosition newSource |
+
+	| errorNode errorMessage errorPosition newSource token |
+	token := currentToken.
+	self step.
```

Comment:
Do not always consume tokens here. parserError means that there is a problem here, not that the current token has to be discarded. We usually want to keep it and instead consider that the current syntactic construction is unfinished.

Reply:
Do you know in what case I should step or not step?

Reply:
I tried to step in #parseLiteralArrayObject instead but then it does not produce a faulty node anymore. I don't know if this is fine?

Reply:
never, it's not the responsibility of `parserError` to do this kind of decision, but it is the job of its callers that have a better knowledge of the syntactic situation.